### PR TITLE
feat: add benchmarking tool for benchmarking Vega APIs

### PR DESCRIPTION
**Pull request by guoguojin**

### CODE REVIEW

File: benchmark/benchmark.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	sort.Slice(values, func(i, j int) bool {
+		return values[i] < values[j]
+	})
+	middle := len(values) / 2
+	if len(values)%2 == 0 {
+		return (values[middle-1] + values[middle]) / 2
+	}
+
+	middle = len(values) / 2
```
Comment (peterbarrow) [inline benchmark/benchmark.go]:
```suggestion
```

File: benchmark/benchmark.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package benchmark
```
Comment (peterbarrow) [inline benchmark/benchmark.go]: Can we call this something more descriptive than just benchmark?
Maybe grpcApiBenchmark or something similar.

File: benchmark/list_orders.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+func listOrders(command *cobra.Command, args []string) {
+	parsePersistentFlags(command)
+
+	log.Printf("Benchmarking ListOrders for: %v", urls)
+	log.Print(", Iteration, Host, mean, median, latency, latency std dev")
+
+	for _, url := range urls {
+		conn, err := grpc.Dial(url, grpc.WithTransportCredentials(insecure.NewCredentials()), ratelimit.WithSecret())
+		if err != nil {
+			log.Printf("could not corrent to %s, %v", url, err)
```
Comment (peterbarrow) [inline benchmark/list_orders.go]:
```suggestion
			log.Printf("could not connect to %s, %v", url, err)
```

File: cmd/benchmark.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+var benchmarkCmd = &cobra.Command{
+	Use:   "benchmark",
+	Short: "A benchmarking tool for vega APIs",
+}
+
+func init() {
+	rootCmd.AddCommand(benchmarkCmd)
+	benchmarkCmd.PersistentFlags().StringSliceP("urls", "u", []string{"localhost:3007"}, "gRPC host:port of the server to benchmark")
```
Comment (peterbarrow) [inline cmd/benchmark.go]:
```suggestion
	benchmarkCmd.PersistentFlags().StringSliceP("urls", "u", []string{"localhost:3007"}, "list of gRPC host:port of the servers to benchmark")
```

Comment (guoguojin) [reply cmd/benchmark.go]: addressed all review comments in https://github.com/vegaprotocol/vegatools/pull/277/commits/70c156fa5dfa7f9f6de0644449cf0114c94175a4

**Review by peterbarrow — APPROVED** [review]

**Pull request merged**